### PR TITLE
Address clippy lints

### PR DESCRIPTION
**Pull request by robjtede**

### CODE REVIEW

**Comment by jaymell** [comment]
Can we instead just fix the errors as suggested by Clippy?

**Comment by robjtede** [comment]
Two of them were fixable, the other 3 are false positives from Clippy; they cannot be fixed in the naïve way since their initial values are `Err(_)`.

File: redis/src/aio/connection.rs [hunk 1]
```diff
@@ -386,6 +386,7 @@ pub(crate) async fn connect_simple<T: RedisRuntime>(
     Ok(match connection_info.addr {
         ConnectionAddr::Tcp(ref host, port) => {
             let socket_addrs = get_socket_addrs(host, port).await?;
+            #[allow(clippy::manual_try_fold)]
```
Comment (jaymell) [inline redis/src/aio/connection.rs:389]: I think the two instances in this file should probably be refactored to not use a fold, honestly. I don't think there's any reason to evaluate all the potential addresses, so we can probably just do something like this instead:

```
future::select_ok(
    socket_addrs.into_iter().map(|socket_addr| {
        Box::pin(async move { <T>::connect_tcp(socket_addr).await })
    })).await?.0
```

Wanna try it?

File: redis/tests/test_cluster_async.rs [hunk 2]
```diff
@@ -175,13 +174,10 @@ async fn test_failover(env: &TestClusterContext, requests: i32, value: i32) {
                         .iter_mut()
                         .map(do_failover)
                         .collect::<stream::FuturesUnordered<_>>()
-                        .fold(
-                            Err(anyhow::anyhow!("None")),
-                            |acc: Result<(), _>, result: Result<(), _>| async move {
-                                acc.or(result)
-                            },
-                        )
+                        .filter_map(|map| future::ready(map.ok()))
```
Comment (jaymell) [inline redis/tests/test_cluster_async.rs:177]: I'm not sure this change is appropriate. If I'm understanding this correctly, it changes the behavior a bit, doesn't it? If I'm not mistaken, with your change, the first success will cause early termination, whereas the original goal of the test was to issue the command on all the nodes. Personally, I'm fine w/ just leaving the override in place here as you had previously.

**Comment by jaymell** [comment]
Issue fixed by #937

**Pull request closed without merging**